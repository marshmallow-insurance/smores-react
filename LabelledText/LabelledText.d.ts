import { FC, ReactNode } from 'react';
import { MarginProps } from '../utils/space';
export type LabelledTextProps = {
    children: ReactNode;
    label: string;
    labelMargin?: 2 | 4 | 8;
} & MarginProps;
export declare const LabelledText: FC<LabelledTextProps>;
