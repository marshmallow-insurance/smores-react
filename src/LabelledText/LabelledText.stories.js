import React from 'react'
import { LabelledText } from './LabelledText'

export default {
  title: 'LabelledText',
  component: LabelledText,
}

const Template = (args) => <LabelledText {...args}>Hideo Kojima</LabelledText>

export const Default = Template.bind({})

Default.args = {
  label: 'Name',
}
