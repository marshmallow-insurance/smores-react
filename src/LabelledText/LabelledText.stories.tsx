import React from 'react'
import { LabelledText, LabelledTextProps } from './LabelledText'

export default {
  title: 'LabelledText',
  component: LabelledText,
}

const Template = (props: LabelledTextProps) => (
  <LabelledText {...props}>Hideo Kojima</LabelledText>
)

export const Default = Template.bind({})

Default.args = {
  label: 'Name',
}
