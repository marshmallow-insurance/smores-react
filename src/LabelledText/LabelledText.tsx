import React, { FC, JSX, ReactNode } from 'react'
import styled from 'styled-components'
import { MarginProps } from '../utils/space'

import { Box } from '../Box'
import { Text } from '../Text'

export type LabelledTextProps = {
  children: ReactNode
  label: string | JSX.Element
  labelMargin?: number
} & MarginProps

export const LabelledText: FC<LabelledTextProps> = ({
  label,
  children,
  labelMargin = 4,
  ...marginProps
}) => (
  <Container {...marginProps} $labelMargin={labelMargin + 'px'}>
    <Text tag="label" color="sesame" typo="label">
      {label}
    </Text>
    {children}
  </Container>
)

const Container = styled(Box)<{ $labelMargin: string }>`
  display: flex;
  flex-direction: column;

  label {
    margin-bottom: ${({ $labelMargin }) => $labelMargin};
  }
`
