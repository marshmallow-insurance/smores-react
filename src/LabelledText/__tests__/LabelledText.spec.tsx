import { render } from '@testing-library/react'
import React from 'react'

import { LabelledText } from '../LabelledText'

test('renders', () => {
  const { container } = render(
    <LabelledText label="Name">Hideo Kojima</LabelledText>,
  )
  expect(container.firstChild).toMatchSnapshot()
})
