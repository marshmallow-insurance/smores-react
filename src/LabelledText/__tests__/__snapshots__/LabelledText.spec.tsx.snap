// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders 1`] = `
<div
  class="sc-aXZVg sc-dcJsrY byxNQM STiR"
>
  <label
    class="sc-aXZVg sc-fqkvVR byxNQM loBhWK"
    color="sesame"
    cursor="inherit"
    title=""
  >
    Name
  </label>
  Hideo Kojima
</div>
`;
